jupyter-extension/src/plugins/mount/mountDrive.ts

```diff
@@ -26,70 +32,76 @@ const DEFAULT_CONTENT_MODEL: Contents.IModel = {
 };
 
 export class MountDrive implements Contents.IDrive {
-  _registry: DocumentRegistry;
+  // Properties necessary for implementing IDrive, but otherwise unused.
+  readonly _registry: DocumentRegistry;
   modelDBFactory?: ModelDB.IFactory | undefined;
-  readonly _model: Paging.IModel;
-
   private _fileChanged = new Signal<this, Contents.IChangedArgs>(this);
-  private _loading = new Signal<this, boolean>(this);
   private _serverSettings = ServerConnection.makeSettings();
   private _isDisposed = false;
+
+  // Signal emits `true` on navigating to a new directory and `false` on response.
+  private _loading = new Signal<this, boolean>(this);
+  // Contents cache for the Drive used to avoid the small Contents limit that can be visible in a FileBrowser.
   private _cache: {
     key: string | null;
     now: number | null;
     contents: Contents.IModel[];
+    filteredContents: Contents.IModel[];
```

Comment:
Just for clarification, why this addition to the cache?

Reply:
`_cache.contents` refers to all the files for the current directory whereas `_cache.filteredContents` refers to the contents currently filtered by the value the user has typed into the FilterBox above the FileBrowser.

Since `_get` is called each time the user changes the `_index` by scrolling, we want to avoid filtering the cached contents each call since the cached contents could be huge. Thus the filtering only occurs when the user changes directory or changes the filtering then is cached.

Reply:
Ahh ok, the piece I was missing was that _get is called for every scroll. thanks!